src/main.rs

```diff
@@ -0,0 +1,246 @@
+use std::{
+    env, fs,
+    io::{BufRead, Write},
+    path::{Path, PathBuf},
+};
+
+use anyhow::{Context, Result};
+use log::info;
+use sequoia_openpgp::{policy::StandardPolicy, serialize::SerializeInto};
+use sq_sdkms::{PgpAgent, SupportedPkAlgo};
+use structopt::StructOpt;
+
+const ENV_API_KEY: &str = "FORTANIX_API_KEY";
+const ENV_API_ENDPOINT: &str = "FORTANIX_API_ENDPOINT";
+
+#[derive(StructOpt)]
+/// OpenPGP integration for Fortanix SDKMS
+struct Cli {
+    /// .env file containing SQ_SDKMS_API_KEY, SQ_SDKMS_API_ENDPOINT
+    #[structopt(long, parse(from_os_str))]
+    env_file: Option<PathBuf>,
+    #[structopt(subcommand)]
+    cmd:      Command,
+}
+
+#[derive(StructOpt)]
+enum Command {
+    /// Produces a detached signature of the given file with SDKMS
+    SignDetached {
+        #[structopt(flatten)]
+        args: CommonArgs,
+        #[structopt(parse(from_os_str))]
+        file: PathBuf,
+    },
+    /// Decrypts the given file with SDKMS, using Sequoia's standard policy
+    Decrypt {
+        #[structopt(flatten)]
+        args: CommonArgs,
+        #[structopt(parse(from_os_str))]
+        file: PathBuf,
+    },
+    /// Generates a PGP key in SDKMS, and outputs the Transferable Public Key
+    GenerateKey {
+        #[structopt(flatten)]
+        args:    CommonArgs,
+        /// An RFC2822-compliant user ID (e.g., "Paul Morphy
+        /// <paul@fortanix.com>")
+        #[structopt(long, short)]
+        user_id: Option<String>,
```

Comment:
Don't you think using `user-id` could cause a confusion with the user `uuid`?
Would it be better to rename it to `user`.

Reply:
I thought about this one. User ID is the standard way of referring to that string and the packet (see RFC4880bis and RFC2822). Changing it to `user` means that the CLI would have `--user` command, which might create confusion with the user of the sdkms API, whereas `uuid` is more general and refers to universal identifiers. As this API does not use user auth, it should be clear to the user that they need to put a complete (RFC2822) `--user-id` as an option. In any case, this option is optional and the CLI guides the user through the process of creating the user ID.